### PR TITLE
Housekeeping: Move BloC logic into use cases and some other refactoring

### PR DESCRIPTION
**Pull request by matthiasbruns**
To get to a cleaner code, we need to do some housekeeping and move files around to keep the refactoring per pr as small as possible.

I am aiming for a classic clean architecture 

UI -> Bloc -> Domain <- Data <- Api/DB

This will lead to
- less static classes
- clear call hierarchy
- testability

### CODE REVIEW

**Comment by oianmol** [comment]
Uncle BOB will be very happy 😄

File: lib/bloc/notification/ct_notification_bloc.dart [hunk 1]
```diff
@@ -29,21 +33,22 @@ class CTNotificationsBloc
...
-      {int pageNo = 1}) async* {
+      {int pageNo = 0}) async* {
     yield state.rebuild((b) => b
       ..loading = true
       ..pageNo = pageNo);
 
     try {
-      var notifications = await ApiRepository.getNotificationsList(pageNo);
+      var notifications =
+          await _fetchNotificationsUseCase.execute(pageIndex: pageNo);
```
Comment (oianmol) [inline lib/bloc/notification/ct_notification_bloc.dart]: @matthiasbruns page no can't be 0

Comment (matthiasbruns) [reply lib/bloc/notification/ct_notification_bloc.dart]: the use case handles the offset

File: lib/domain/notification/usecase/fest_notifications.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+class FetchNotificationsUseCase {
+  // TODO add DI to handle this for us
+  final ApiRepository _repository = ApiRepository.instance;
+
+  /// Will return a [Future] of type [ResponseNotifications] when called.
+  ///
+  /// Optional [pageIndex] starting at 0 for paginated notification lists.
+  Future<ResponseNotifications> execute({int pageIndex = 0}) async {
+    return _repository.getNotificationsList(pageIndex + 1);
```
Comment (oianmol) [inline lib/domain/notification/usecase/fest_notifications.dart]: @matthiasbruns why not sending exact page no ?

Comment (matthiasbruns) [reply lib/domain/notification/usecase/fest_notifications.dart]: FetchNotificationsUseCase line 12

File: lib/network/notification/response_notification.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  String _message;
+
+  List<ResponseNotificationItem> get data => _data;
+
+  String get message => _message;
+
+  ResponseNotifications(this._data, this._message);
+
+  ResponseNotifications.map(dynamic obj) {
+    print(obj);
```
Comment (oianmol) [inline lib/network/notification/response_notification.dart:16]: @matthiasbruns lets remove all print or add logging lib

File: lib/domain/notification/usecase/fest_notifications.dart (hunk 2 above)
Comment (matthiasbruns) [reply lib/domain/notification/usecase/fest_notifications.dart]: It feels weird as a caller to not start at 0 when working with indexes.

File: pubspec.lock [hunk 4]
```diff
@@ -161,7 +161,7 @@ packages:
       name: crypto
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.1.4"
+    version: "2.1.3"
```
Comment (oianmol) [inline pubspec.lock]: @matthiasbruns I see all downgraded version, any reason ?

File: lib/network/notification/response_notification.dart (hunk 3 above)
Comment (matthiasbruns) [reply lib/network/notification/response_notification.dart:16]: https://coronatrace.atlassian.net/browse/DEV-118

File: lib/domain/notification/usecase/fest_notifications.dart (hunk 2 above)
Comment (matthiasbruns) [reply lib/domain/notification/usecase/fest_notifications.dart]: This feels to my like hidden business logic, which should not be handled be the calling parties

Comment (oianmol) [reply lib/domain/notification/usecase/fest_notifications.dart]: @matthiasbruns these are no arrays to use the term indexes, these are pages, and should not be 0 .

File: lib/bloc/notification/ct_notification_bloc.dart (hunk 1 above)
Comment (oianmol) [reply lib/bloc/notification/ct_notification_bloc.dart]: where are the page no's getting incremented ?

File: lib/domain/notification/usecase/fest_notifications.dart (hunk 2 above)
Comment (matthiasbruns) [reply lib/domain/notification/usecase/fest_notifications.dart]: done

File: pubspec.lock (hunk 4 above)
Comment (matthiasbruns) [reply pubspec.lock]: error while rebasing. Give me a sec

Comment (matthiasbruns) [reply pubspec.lock]: I need to find out why my cli keep downgrading the dependencies

File: lib/bloc/notification/ct_notification_bloc.dart (hunk 1 above)
Comment (matthiasbruns) [reply lib/bloc/notification/ct_notification_bloc.dart]: line 29

```kotlin
      case NotificationsBloEvent.fetchMore:
        yield* _fetchNotificationsInternal(pageNo: state.pageNo + 1);
        break;

```

**Comment by oianmol** [comment]
@matthiasbruns Please resolve conflicts and I will merge your PR.

**Comment by matthiasbruns** [comment]
@Anmol92verma was busy working :D done

**Pull request merged**